content/en/docs/languages/net/instrumentation.md

```diff
@@ -22,39 +35,219 @@ are covered here as well as the `System.Diagnostics` API.
+
+    public DiceController(ILogger<DiceController> logger)
+    {
+        this.logger = logger;
+    }
+
+    [HttpGet("/rolldice")]
+    public List<int> RollDice(string player, int? rolls)
+    {
+        if(!rolls.HasValue)
+        {
+            logger.LogError("Missing rolls parameter");
+            throw new HttpRequestException("Missing rolls parameter", null, HttpStatusCode.BadRequest);
+        }
+            
+        var result = new Dice(1, 6).rollTheDice(rolls.Value);
+
+        if (string.IsNullOrEmpty(player))
+        {
+            logger.LogInformation("Anonymous player is rolling the dice: {result}", result);
+        }
+        else
+        {
+            logger.LogInformation("{player} is rolling the dice: {result}", player, result);
+        }
+
+        return result;
+    }
+}
+
+```
+
+Replace the content of the Program.cs file with the following code:
+
+```csharp
+var builder = WebApplication.CreateBuilder(args);
+
+builder.Services.AddControllers();
+
+var app = builder.Build();
+
+app.MapControllers();
+
+app.Run();
+```
+
+To ensure that it is working, run the application with the following command and
+open <http://localhost:8080/rolldice?rolls=12> in your web browser:
+
+```sh
+dotnet run
+```
+
+You should get a list of 12 numbers in your browser window, for example:
+
+```text
+[5,6,5,3,6,1,2,5,4,4,2,4]
+```
+
+## Manual instrumentation setup
+
+### Dependencies
+
+Install the following OpenTelemetry NuGet packages:
+
+[OpenTelemetry.Exporter.Console](https://www.nuget.org/packages/OpenTelemetry.Exporter.Console)
+
+[OpenTelemetry.Extensions.Hosting](https://www.nuget.org/packages/OpenTelemetry.Extensions.Hosting)
 
-First, ensure that you have the right packages:
 
 ```sh
-dotnet add package OpenTelemetry
 dotnet add package OpenTelemetry.Exporter.Console
+dotnet add package OpenTelemetry.Extensions.Hosting
 ```
 
-And then use code like this at the beginning of your program, during any
-important startup operations.
+For ASP.NET Core-based applications, also install the AspNetCore instrumentation package
+
+[OpenTelemetry.Instrumentation.AspNetCore](https://www.nuget.org/packages/OpenTelemetry.Instrumentation.AspNetCore)
+
+```sh
+dotnet add package OpenTelemetry.Instrumentation.AspNetCore
+```
+
+
+### Initialize the SDK
+
+{{% alert title="Note" color="info" %}} If you’re instrumenting a library, you dont need to initialize the SDK. {{% /alert %}}
+
+It is important to configure an instance of the OpenTelemetry SDK as early as possible in your application.
+
+To initialize the OpenTelemetry SDK for an ASP.NET Core app like in the case of the example app, 
+replace the content of the program.cs file with the following code: 
+
+```csharp
+using OpenTelemetry.Logs;
+using OpenTelemetry.Metrics;
+using OpenTelemetry.Resources;
+using OpenTelemetry.Trace;
+
+// Ideally, you will want this name to come from a config file, constants file, etc.
+var serviceName = "Dice.*";
```

Comment:
```suggestion
var serviceName = "dice-server";
```

Reply:
@svrnm if we change this "dice-server" then the name to create the ActivitySource for both the server and the lib will have to be "dice-server" as well.

Reply:
let me double check on that, might be that .NET behaves here differently compared to other languages

Reply:
Ok... so if I understand it correctle then, the ActivitySource name should be the instrumentation scope name, not the service name, so they should be different.

Reply:
@open-telemetry/dotnet-approvers please take a look, I am no expert on that matter here, might be my misunderstanding

Reply:
ActivitySource == Tracer
Activity == Span

Since it's normally 1 tracer per app, I'd say `dice-server` is fine?